check/checksum.go

```diff
@@ -0,0 +1,182 @@
+		return nil, fmt.Errorf("error reading %s content: %w", path, err)
+	}
+
+	return h.Sum(nil), nil
+}
+
+func checkChecksum(src, target string) (bool, error) {
+	name := filepath.Base(src)
+	sc, err := checksumFor(src)
+	if err != nil {
+		return false, fmt.Errorf("error getting %s checksum: %w", src, err)
+	}
+
+	targetPath := filepath.Join(target, name)
+	tc, err := checksumFor(targetPath)
+	if err != nil {
+		return false, fmt.Errorf("error getting %s checksum: %w", targetPath, err)
+	}
+
+	return bytes.Equal(sc, tc), nil
+}
+
+// Check compares md5 checksum files from src directory with corresponding files in target directory
+func CheckChecksum(src, target string) error {
+	ls, err := filepath.Glob(filepath.Join(src, "*.md5"))
+	if err != nil {
+		return fmt.Errorf("error listing checksum files in %s directory: %w", src, err)
+	}
+
+	if len(ls) == 0 {
+		return fmt.Errorf("target directory %s has no checksum files to compare with", target)
+	}
+
+	bar := progressbar.Default(int64(len(ls)))
+	defer bar.Close()
+	bar.Describe("Checking files checksum")
+	if err := bar.RenderBlank(); err != nil {
+		return fmt.Errorf("error rendering the progress bar: %w", err)
+	}
+
+	type qm struct {
+		f     string
+		equal bool
+		err   error
+	}
+	q := make(chan qm, len(ls))
+	defer close(q)
+
+	var wg sync.WaitGroup
+
+	for _, f := range ls {
+		wg.Add(1)
+
+		go func(f string, wg *sync.WaitGroup, q chan<- qm) {
+			defer wg.Done()
+
+			equal, err := checkChecksum(f, target)
+
+			q <- qm{f, equal, err}
+		}(f, &wg, q)
+	}
+
+	wg.Wait()
+
+	var notEqual []string
+	for m := range q {
+		if m.err != nil {
+			return err
+		}
+
+		bar.Add(1)
+
+		if bar.IsFinished() {
+			if len(notEqual) > 0 {
+				return fmt.Errorf("got different checksum for file(s): %v", notEqual)
+			}
+
+			log.Output(1, "OK!")
+			return nil
+		}
+
+		if !m.equal {
+			notEqual = append(notEqual, m.f)
+		}
+	}
+
+	return nil
+}
+
+func createChecksum(src string, f string) error {
+	checksum, err := checksumFor(filepath.Join(src, f))
+	if err != nil {
+		return fmt.Errorf("error getting checksum for %s: %w", src, err)
+	}
+
+	targetPath := filepath.Join(src, fmt.Sprintf("%s.md5", f))
+	if err := os.WriteFile(targetPath, checksum, 0755); err != nil {
+		return fmt.Errorf("error writing %s checksum file: %w", targetPath, err)
+	}
+
+	return nil
+}
+
+// Create creates a md5 checksum file in target directory for each file in src directory
+func CreateChecksum(src string) error {
+	ls, err := os.ReadDir(src)
+	if err != nil {
+		return fmt.Errorf("error reading %s directory: %w", src, err)
+	}
+
+	bar := progressbar.Default(int64(len(ls)))
+	defer bar.Close()
+	bar.Describe("Creating checksum files")
+	if err := bar.RenderBlank(); err != nil {
+		return fmt.Errorf("error rendering the progress bar: %w", err)
+	}
+
+	q := make(chan error, len(ls))
+	defer close(q)
+
+	var wg sync.WaitGroup
+
+	for _, f := range ls {
+		if f.IsDir() || isHiddenFile(f.Name()) {
```

Comment:
Se a gente só usa `isHiddenFile` aqui, acho que não precisamos dela

```suggestion
		if f.IsDir() || f.Name()[0] == '.' {
```